src/Sylius/Bundle/CoreBundle/PaymentRequest/Checker/PaymentRequestIntegrityChecker.php

```diff
@@ -29,16 +30,19 @@ public function __construct(
     public function check(PaymentRequestHashAwareInterface $command): PaymentRequestInterface
     {
         $hash = $command->getHash();
-        Assert::notNull($hash, 'Payment request hash cannot be null.');
 
         $paymentRequest = $this->paymentRequestRepository->findOneByHash($hash);
-        Assert::notNull($paymentRequest, sprintf('Payment request (hash "%s") not found.', $hash));
+        if (null === $paymentRequest) {
+            throw new PaymentRequestNotFoundException();
+        }
 
         /** @var PaymentMethodInterface $paymentMethod */
         $paymentMethod = $paymentRequest->getMethod();
 
         $gatewayConfig = $paymentMethod->getGatewayConfig();
-        Assert::notNull($gatewayConfig, 'Gateway config cannot be null.');
+        if (null === $gatewayConfig) {
+            throw new NullGatewayConfigException();
+        }
 
         return $paymentRequest;
```

Comment:
`Checker` shouldn't return anything besides true/false.

Reply:
Yeah, I will rename it to provider, as it's actually used to provide PaymentRequestS